pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -505,10 +505,14 @@ func SecretToLibvirtSecret(vm *v1.VirtualMachine, secretName string) string {
 	return fmt.Sprintf("%s-%s-%s---", secretName, vm.Namespace, vm.Name)
 }
 
-func QuantityToMegaByte(quantity resource.Quantity) Memory {
+func QuantityToByte(quantity resource.Quantity) Memory {
+	memorySize, _ := quantity.AsInt64()
+	if memorySize < 0 {
```

Comment:
When can you receive here negative value?

Reply:
I have added this condition because ParseQuantity accepts negative values, for example:
```ParseQuantity("-64Mi")``` aftter conversion to ```Int64``` i will have value ```-67108864``` and then there is cast to ```uint64``` which gives us result ```18446744073642442752```.

Being honest I don't expect users to specify negative memory values for vm :), but there was this possibility and that's why i added this edge condition, just to be safe.

Reply:
Uh, interesting if k8s accept negative values.

Reply:
hmm i checked only this ```ParseQuantity``` function, maybe they have some validation at some level, I will check real scenario with it.

Reply:
Can you add a check for negative quantities in `pkg/virt-api/validating-webhook/validating-webhook.go`? If we would see negative values here, I would expect that virt-handler would reject starting the vm.

Reply:
Sure I will do this.

Reply:
Also remove the negative number flip. It should be enought to return an error from the converter with a reason able error message.

Reply:
Done

Reply:
Oh I said done, but it's not, I will return error from the converter... i just removed this flip... ok I am fixing this